demo/admin/src/products/ManufacturerForm.tsx

```diff
@@ -0,0 +1,376 @@
+import { useApolloClient, useQuery } from "@apollo/client";
+import {
+    Field,
+    FieldSet,
+    FinalForm,
+    FinalFormInput,
+    FinalFormSaveSplitButton,
+    FinalFormSubmitEvent,
+    FormSection,
+    Loading,
+    MainContent,
+    TextField,
+    Toolbar,
+    ToolbarActions,
+    ToolbarFillSpace,
+    ToolbarItem,
+    ToolbarTitleItem,
+    useFormApiRef,
+    useStackApi,
+    useStackSwitchApi,
+} from "@comet/admin";
+import { ArrowLeft } from "@comet/admin-icons";
+import { EditPageLayout, queryUpdatedAt, resolveHasSaveConflict, useFormSaveConflict } from "@comet/cms-admin";
+import { IconButton } from "@mui/material";
+import { FormApi } from "final-form";
+import { filter } from "graphql-anywhere";
+import isEqual from "lodash.isequal";
+import React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { createManufacturerMutation, manufacturerFormFragment, manufacturerQuery, updateManufacturerMutation } from "./ManufacturerForm.gql";
+import {
+    GQLCreateManufacturerMutation,
+    GQLCreateManufacturerMutationVariables,
+    GQLManufacturerFormDetailsFragment,
+    GQLManufacturerQuery,
+    GQLManufacturerQueryVariables,
+    GQLUpdateManufacturerMutation,
+    GQLUpdateManufacturerMutationVariables,
+} from "./ManufacturerForm.gql.generated";
+
+type FormValues = Omit<GQLManufacturerFormDetailsFragment, "address" | "addressAsEmbeddable"> & {
+    address: {
+        street: string;
+        streetNumber: string | null;
+        zip: string;
+        country: string;
+        alternativeAddress: { street: string; streetNumber: string | null; zip: string; country: string };
+    };
+    addressAsEmbeddable: {
+        street: string;
+        streetNumber: string | null;
+        zip: string;
+        country: string;
+        alternativeAddress: { street: string; streetNumber: string | null; zip: string; country: string };
+    };
+};
+
+interface FormProps {
+    id?: string;
+}
+
+export function ManufacturerForm({ id }: FormProps): React.ReactElement {
+    const stackApi = useStackApi();
+    const client = useApolloClient();
+    const mode = id ? "edit" : "add";
+    const formApiRef = useFormApiRef<FormValues>();
+    const stackSwitchApi = useStackSwitchApi();
+
+    const { data, error, loading, refetch } = useQuery<GQLManufacturerQuery, GQLManufacturerQueryVariables>(
+        manufacturerQuery,
+        id ? { variables: { id } } : { skip: true },
+    );
+
+    const initialValues = React.useMemo<Partial<FormValues>>(
+        () =>
+            data?.manufacturer
+                ? {
+                      ...filter<GQLManufacturerFormDetailsFragment>(manufacturerFormFragment, data.manufacturer),
+                      address: data.manufacturer.address
+                          ? {
+                                street: data.manufacturer.address.street,
```

Comment:
you should also use spread operator in nested objects for all text fields

Reply:
if I use the spread operator for initialValues (like the comment line I added) `__typename` (part of graphql query response) is also set for initialValues and sent with mutation input (if spread operator is also used there) resulting in an graphql error as `__typename` is not expected to be part of the input type.

so either we use spread operator at least in only one of this two places to get rid of `__typename` or use some kind of filter (like in [line 79](https://github.com/vivid-planet/comet/pull/1751/files#diff-62e570ff3ed31759613e194bc4fca59606ed16d03e455d24f6b20054c1283957R79)) to remove unused fields. I don't know if there is any handy js-operator-daisychaining-trick to remove a field from an object.

Reply:
It's odd that we can't use `filter` for `address`, but this is probably due to the number fields...

Reply:
@nsams how should I proceed with this concern? it's working in the current state because I'm not using the spread-operator for initialValues.

Reply:
@nsams found a solution. moved the complete object creation into the second-parameter of filter. now __typename is also removed from nested fields

Reply:
The fragments and the `filter` function both should work correctly with nested fields.

Reply:
the problem is with update-mutation failing when __typename is sent. and __typename does not make sense if it's included in initialValues